delphi/translators/for2py/genPGM.py

```diff
@@ -222,13 +222,29 @@ def genPgm(self, node, state, fnNames, call_source):
             loopFn = {
                 "name": loopName,
                 "type": "loop_plate",
-                "input": variables,
+                "input": [
+                    {
+                      "name": variable,
+                      "domain": state.varTypes[variable]
+                    }
+                    for variable in variables
+                ],
                 "index_variable": indexName,
                 "index_iteration_range": iterationRange,
                 "body": loopBody,
             }
 
-            loopCall = {"name": loopName, "inputs": variables, "output": {}}
+            loopCall = {
+                "name": loopName,
+                "input": [
+                    {
+                      "name": variable,
+                      "index": -1,
```

Comment:
This line can probably just be changed to `"domain": state.varTypes[variable]` for now